pymatgen/io/pwscf.py

```diff
@@ -334,10 +335,12 @@ def input_mode(line):
                         coords_are_cartesian = True
                     elif mode[1] == "crystal":
                         coords_are_cartesian = False
+
         structure = Structure(
             Lattice(lattice),
             species,
             coords,
+            # DEBUG (@DanielYang59): need input on coords_are_cartesian
```

Comment:
Need help on `io.pwscf` @janosh (I never use `pwscf`): `coords_are_cartesian` is only defined in one of the many branches, and is unbound elsewhere.

Reply:
wow, so many code branches and only one of them was actually tested?

https://github.com/materialsproject/pymatgen/blob/37d1066a8f45b24dff8cf8c18b2549648897afce/tests/io/test_pwscf.py#L360-L367

this seems in urgent need of more tests. i think defaulting `coords_are_cartesian=False` seems right. i also never used this code so it's hard to know for sure. apparently no one does or this would have been reported long ago

Reply:
> so many code branches and only one of them was actually tested?

Yes, `coords_are_cartesian` is defined in only one of the few branches and is unbound all elsewhere 😓, meaning the other branches were likely never entered.